.gitlab/kernel_version_testing/common.yml

```diff
@@ -0,0 +1,285 @@
+  tags: ["arch:amd64"]
+  allow_failure: true
+  script:
+    # Build dependencies directory
+    - mkdir -p $DEPENDENCIES
+    - pushd $DEPENDENCIES
+    # download and copy btf files
+    - mkdir -p $BTF_DIR
+    - cp $CI_PROJECT_DIR/minimized-btfs.tar.xz $BTF_DIR/minimized-btfs.tar.xz
+    # package all the dependencies
+    - ls -la $DEPENDENCIES
+    - pushd $DD_AGENT_TESTING_DIR/$ARCH
+    - tar czvf ../$ARCHIVE_NAME btfs
+    - popd
+    # upload tests
+    # Switch to new profile after the btfs have been downloaded. Switching before
+    # causes permission issues.
+    - !reference [.kernel_matrix_testing_new_profile]
+    - !reference [.write_ssh_key_file]
+    - !reference [.wait_for_instances]
+    - !reference [.get_instance_ip_by_type]
+    - !reference [.setup_ssh_config]
+    - scp $DD_AGENT_TESTING_DIR/$ARCHIVE_NAME metal_instance:/opt/kernel-version-testing/
+  variables:
+    DEPENDENCIES: $DD_AGENT_TESTING_DIR/$ARCH/btfs
+    BTF_DIR: opt/system-probe-tests/pkg/ebpf/bytecode/build/co-re/btf
+    AWS_EC2_SSH_KEY_FILE: $CI_PROJECT_DIR/ssh_key
+
+# -- Environment setup
+.kernel_matrix_testing_setup_env:
+  extends:
+    - .kitchen_ec2_location_us_east_1
+  stage: kernel_matrix_testing_prepare
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/test-infra-definitions/runner$TEST_INFRA_DEFINITIONS_BUILDIMAGES_SUFFIX:$TEST_INFRA_DEFINITIONS_BUILDIMAGES
+  needs: ["go_deps", "go_tools_deps"]
+  tags: ["arch:amd64"]
+  variables:
+    AWS_REGION: us-east-1
+    STACK_DIR: $CI_PROJECT_DIR/stack.dir
+    # The ssh key is created by the pulumi scenario, to be used for creating
+    # instances in the build-stable account. We reuse this file to ssh into
+    # the instances in subsequent jobs.
+    AWS_EC2_SSH_KEY_FILE: $CI_PROJECT_DIR/ssh_key
+    AWS_EC2_SSH_KEY_NAME: datadog-agent-ci
+    INFRA_ENV: "aws/agent-qa"
+    PIPELINE_ID: $CI_PIPELINE_ID
+    TEAM: "ebpf-platform"
+    RESOURCE_TAGS: "instance-type:${INSTANCE_TYPE},arch:${ARCH},test-component:${TEST_COMPONENT}"
+    KUBERNETES_MEMORY_REQUEST: "12Gi"
+    KUBERNETES_MEMORY_LIMIT: "16Gi"
+    VMCONFIG_FILE: "${CI_PROJECT_DIR}/vmconfig-${CI_PIPELINE_ID}-${ARCH}.json"
+    TEST_SETS: "no_tracersuite,only_tracersuite,all_tests"
+  before_script:
+    - set +x
+    - export DD_API_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.datadog_api_key_org2 --with-decryption --query "Parameter.Value" --out text)
+    - set -x
+    - !reference [.retrieve_linux_go_deps]
+    - !reference [.kernel_matrix_testing_new_profile]
+    - !reference [.write_ssh_key_file]
+  script:
+    - echo "s3://dd-pulumi-state?region=us-east-1&awssdk=v2&profile=$AWS_PROFILE" > $STACK_DIR
+    - pulumi login $(cat $STACK_DIR | tr -d '\n')
+    - inv -e kmt.gen-config --ci --arch=$ARCH --output-file=$VMCONFIG_FILE --sets=$TEST_SETS
```

Comment:
We should have separate input template files for `security-agent` and `system-probe` so we do not download the docker disk when it is not needed.

Reply:
Done! Added a `--vmconfig-template` argument to allow changing files (e.g., setting --vmconfig-template security-agent selects `test/new-e2e/system-probe/config/vmconfig-security-agent.json`)